src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,169 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode;
+        if (size == 0) {
+            newNode = new Node<>(null, value, null);
+            head = newNode;
+        } else {
+            newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> newNode;
+        if (index == size) {
+            add(value);
+            return;
+        } else {
+            if (index == 0) {
+                newNode = new Node<>(null, value, head);
+                head.prev = newNode;
+                head = newNode;
+            } else {
+                checkForValidIndex(index);
+                Node<T> next = getNodeByIndex(index - 1);
+                Node<T> prev = next.prev;
+                newNode = new Node<>(prev, value, next);
+                newNode.next = next.next;
+                next.next = newNode;
+                newNode.prev = next;
+                newNode.next.prev = newNode;
+            }
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
+        checkForValidIndex(index);
+        Node<T> newNode = head;
+        int countIndex = 0;
+        while (newNode != null) {
+            if (countIndex == index) {
+                return (T) newNode.value;
+            }
+            countIndex++;
+            newNode = newNode.next;
+        }
         return null;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> newNode;
+        checkForValidIndex(index);
+        newNode = getNodeByIndex(index);
```

Comment:
```suggestion
        Node<T> node = getNodeByIndex(index);
```
it's rather old than new